### DISCUSSION TITLE
Vehicle Routing Problem (VRP) with new orders/routes

### DISCUSSION THREAD

**Discussion by pcaruso** [google/or-tools#2971] [Routing (and legacy CP) questions]
Hello everyone. I am trying to develop a multi driver delivery program where orders need to be picked up from a depot, delivered to customers addresses and come back. So far I managed to implement VRP  using OR-TOOLS for the optimum route based on N drivers on duty (see link below). When new orders arrive, drivers always need to come back to the depot after finishing their first complete route to pickup the new order items and deliver them. 
My question is: How to assign new orders to drivers that are still on route considering they will not return and start initially from depot at same time?
I will appreciate any possible help as I am quite lost on how to achieve this for new orders that were not initially calculated.
Thank you all.

Current state: https://youtu.be/mfz7CVpbDN4

**Comment by abduakhatov** [comment]
@pcaruso I had similar case and hope my solutions suits you!
1. You can use [pdp](https://developers.google.com/optimization/routing/pickup_delivery) to make depot pickup point
2. When new order arrives, you can you use #847 to assign picked nodes to vehicle
   Note: picked nodes' locations are set as related vehicles' locations

**Comment by nikhilrajgarhia** [comment]
@abduakhatov do you have some example you implemented with pdp?

**Reply by abduakhatov** [reply]
@nikhilrajgarhia sorry, it has been long time since i posted this answer. Currently, i think i ll not be able to recall or find any example)
